### PR TITLE
clean up some useless fragments after adding type hints to almost eve…

### LINKED ISSUE #78: clean up some type hint related code issues
After implementing type hints for almost every method some code fragments have become useless. For example, checking the return value of a void-method is now no longer neccesary. These should be cleaned up and also some method calls should be made more explicit.

### PR DESCRIPTION
**Pull request by MarcusSchwarz**
…rything; resolves #78

### CODE REVIEW

File: tests/Testhelper/MyCustomSource.php [hunk 1]
```diff
@@ -49,11 +49,11 @@ public function getRealPath(): string
...
 
     /**
-     * @return int
+     * @return float
      */
-    public function getLastModifiedTime(): int
+    public function getLastModifiedTime(): float
     {
-        return $this->path === 'nocache' ? mt_rand() : 0;
+        return (float) ($this->path === 'nocache' ? mt_rand() : 0);
```
Comment (WorksDev) [inline tests/Testhelper/MyCustomSource.php]: I think this schould be int, or?

File: src/StringSource.php [hunk 2]
```diff
@@ -46,16 +46,15 @@ public function __construct($data, ?string $realpath = null)
...
     {
+        $mTime = 0;
+
         if (strpos($this->realpath, self::NO_PATH_PREFIX) !== 0 && file_exists($this->realpath)) {
-            return @filemtime($this->realpath);
+            $mTime = @filemtime($this->realpath);
         }
 
-        return 0;
+        return (float) $mTime;
```
Comment (WorksDev) [inline src/StringSource.php]: filemtime = int, why float?

File: tests/Testhelper/MyCustomSource.php (hunk 1 above)
Comment (MarcusSchwarz) [reply tests/Testhelper/MyCustomSource.php]: https://github.com/SC-Networks/PHPTAL/blob/55761d54b81c0ba972ffb557d7a681f17cfeb994/src/PHPTAL.php#L1053
https://github.com/SC-Networks/PHPTAL/blob/55761d54b81c0ba972ffb557d7a681f17cfeb994/src/PHPTAL.php#L1067
https://github.com/SC-Networks/PHPTAL/blob/55761d54b81c0ba972ffb557d7a681f17cfeb994/src/PHPTAL.php#L916

Comment (MarcusSchwarz) [reply tests/Testhelper/MyCustomSource.php]: Changed to a saner way

**Review by WorksDev — APPROVED** [review]

**Pull request merged**